### PR TITLE
Next.js Demo: UX Improvements 1/2

### PR DESCRIPTION
**Pull request by jmderby**
## Description

This PR comprises of changes from another [PR](https://github.com/Paella-Labs/crossbit-main/pull/14450) and is part 1 of 2.

To test the changes locally, you need to change the following files from the SDK:

1. `auth-core/src/utils/constants.ts`
```
// from this
export const CROSSMINT_STG_URL = "https://staging.crossmint.com";
// to this 
export const CROSSMINT_STG_URL = "http://localhost:3000";
```

2. `Smart Wallet + Auth Demo/package.json`
```
// from this
"dependencies": {
        "@crossmint/client-sdk-react-ui": "1.3.2"
// to this
"dependencies": {
        "@crossmint/client-sdk-react-ui": "workspace:*"
```

### Changelog: 
- X button to close auth modal 
- Can no longer click outside modal to close it and require clicking X
- Demo logo updated to Smarter Wallets 
- Renamed sign in button to "Sign in" 

Screenshot 2024-08-22 at 9 34 54 AM

## Test plan

Demo flow still works as expected.

### CODE REVIEW

File: packages/client/auth-core/src/components/AuthModal.tsx [hunk 1]
```diff
@@ -89,17 +89,48 @@ export default function AuthModal({
...
+                        stroke="currentColor"
+                        strokeWidth="2"
+                        strokeLinecap="round"
+                        strokeLinejoin="round"
+                    >
+                        <path d="M18 6 6 18" />
+                        <path d="m6 6 12 12" />
+                    </svg>
+                </button>
+            </div>
```
Comment (bryan-eastwood3) [inline packages/client/auth-core/src/components/AuthModal.tsx:122]: Is it idiomatic to put this in it's own variable/component somewhere? If not it's fine!

**Review by bryan-eastwood3 — APPROVED** [review]

File: packages/client/auth-core/src/components/AuthModal.tsx (hunk 1 above)
Comment (jmderby) [reply packages/client/auth-core/src/components/AuthModal.tsx:94]: Added it in a separate component below 👍

File: apps/wallets/smart-wallet/next/src/app/template.tsx [hunk 2]
```diff
@@ -30,7 +30,6 @@ export default function Template({ children }: { children: React.ReactNode }) {
     ) : (
         <CrossmintAuthProvider
             apiKey={process.env.NEXT_PUBLIC_CROSSMINT_API_KEY ?? ""}
-            environment="staging"
```
Comment (jmderby) [inline apps/wallets/smart-wallet/next/src/app/template.tsx:33]: I removed this prop as it's no longer being used in a recently merged MR

Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:33]: disregard, reverted.

**Pull request merged**